proto/v3/node_search.proto

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package datacommons.v3;
+
+option go_package = "github.com/datacommonsorg/mixer/internal/proto/v3";
+
+// Node search request sent to mixer.
+message NodeSearchRequest {
+  // Node search query (e.g. "total population", "california").
+  string query = 1;
+
+  // Optional. The types of nodes the search should be restricted to.
+  // e.g. "StatisticalVariable", "Country"
+  // If not specified, nodes of all types will be searched.
+  repeated string types = 2;
+}
+
+// Node search response from mixer.
+message NodeSearchResponse {
+  // List of nodes resulting from the search.
+  repeated NodeSearchResult results = 1;
```

Comment:
@n-h-diaz - The previous [SearchResponse](https://github.com/datacommonsorg/mixer/blob/ec6a2639cdac9d79f516f39f013623597926c2a8/proto/misc.proto#L72) broke down the results by type. In this case, it is modeled as a flat list to maintain ranking order. Let me know your thoughts.